tests/test_render_mcp.py

```
@@ -0,0 +1,596 @@
+        assert MCPClient._parse_body(resp)["result"]["value"] == 42
+
+    def test_sse_without_response_frame_raises(self):
+        resp = _response(
+            body='data: {"jsonrpc":"2.0","method":"notifications/progress"}\n\n',
+            content_type="text/event-stream",
+        )
+        with pytest.raises(ValueError):
+            MCPClient._parse_body(resp)
+
+    def test_accept_header_lists_both_media_types(self):
+        headers = MCPClient("https://example.test", secret_token="tok")._headers()
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to argument: "secret_token"

(S106)